lib/core/sdam/server_description.js

```diff
@@ -119,6 +121,30 @@ class ServerDescription {
   get isWritable() {
     return WRITABLE_SERVER_TYPES.has(this.type);
   }
+
+  /**
+   * Determines if another `ServerDescription` is equal to this one per the rules defined
+   * in the {@link https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#serverdescription|SDAM spec}
+   *
+   * @param {ServerDescription} other
+   * @return {Boolean}
+   */
+  equals(other) {
```

Comment:
any chance of `other` being `undefined` or `null`?

Reply:
hm good call, I'll add in a check for that